node/src/types/block.rs

```diff
@@ -997,6 +997,19 @@ impl BlockSignatures {
     pub(crate) fn has_proof(&self, public_key: &PublicKey) -> bool {
         self.proofs.contains_key(public_key)
     }
+
+    /// Verify the signatures contained within.
+    pub(crate) fn verify(&self) -> crypto::Result<()> {
+        let mut bytes = self.block_hash.inner().to_vec();
+        bytes.extend_from_slice(&self.era_id.0.to_le_bytes());
+        for (key, signature) in self.proofs.iter() {
+            match crypto::verify(bytes.clone(), signature, key) {
+                Ok(_) => continue,
+                Err(error) => return Err(error),
+            }
+        }
```

Comment:
Instead of recreating the signature verification code (which risks doing it incorrectly) we should recreate the `FinalitySignature` and call `verify()` on it:
```rust
let signature = FinalitySignature {
                block_hash: self.block_hash,
                era_id: self.era_id,
                signature: signature.clone(),
                public_key: public_key.clone(),
            };
            match signature.verify() {
                Ok(_) => continue,
                Err(error) => return Err(error),
            }
```

Reply:
Applied suggestion